c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -804,6 +864,14 @@ class DeviceCachingAllocator {
       update_stat(stats.allocation[stat_type], -1);
       update_stat(stats.allocated_bytes[stat_type], -block->size);
     });
+    if (block->history) {
+      record_trace(
+          TraceEntry::FREE,
```

Comment:
I'm uncertain if FREE should be recorded here - true, this is where FREE request happen, but actual freeing might not happen if there are other streams using this block (see the check below), so trace would be confusing - it would imply that memory was freed when in fact it wasn't.